cmd/kube-controller-manager/app/apps.go

```diff
@@ -32,53 +33,53 @@ import (
 	"k8s.io/kubernetes/pkg/controller/statefulset"
 )
 
-func startDaemonSetController(ctx ControllerContext) (http.Handler, bool, error) {
+func startDaemonSetController(ctx context.Context, controllerContext ControllerContext) (http.Handler, bool, error) {
```

Comment:
This introduces unnecessary duplication. `ControllerContext` has `Stop` channel in it. I'm seeing two options to consider:
1. remove/deprecate `Stop` field from `ControllerContext` and pass `context.Context` everywhere
2. inject `context.Context` into `ControllerContext`.

Looking at all the places, including go std lib, the 1st option is widely used and would be my preferred as well.

Reply:
@soltysh sounds good, can we simply pass `ctx.Done()` in each spot the stop channel is passed, or do we need to instantiate a single `ctx.Done()` and pass that through all the way down? Not sure the details